polaris-core/src/main/java/org/apache/polaris/core/PolarisConfiguration.java

```diff
@@ -30,15 +31,23 @@ public class PolarisConfiguration<T> {
   public final T defaultValue;
   private final Optional<String> catalogConfigImpl;
   private final Class<T> typ;
+  private final Optional<Function<T, Boolean>> validation;
 
   @SuppressWarnings("unchecked")
   public PolarisConfiguration(
-      String key, String description, T defaultValue, Optional<String> catalogConfig) {
+      String key,
+      String description,
+      T defaultValue,
+      Optional<String> catalogConfig,
+      Optional<Function<T, Boolean>> validation) {
     this.key = key;
     this.description = description;
     this.defaultValue = defaultValue;
     this.catalogConfigImpl = catalogConfig;
     this.typ = (Class<T>) defaultValue.getClass();
+    this.validation = validation;
+
+    validate(cast(defaultValue));
```

Comment:
Is this just a safety check? Feels like this shouldn't happen right?

Reply:
Also isn't this redundant now. (cast calls validate)

So wouldn't this be 

validate(
   cast(defaultValue){
      validate(defaultValue)
   }
}